shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,77 @@
+import copy
+import json
+import math
+import nibabel as nib
+import numpy as np
+import os
+
+
+def read_nii(nii_path):
+    """ Reads a nifti file and returns image, info and json_data
+
+    Args:
+        nii_path (str): direct path to the .nii or .nii.gz file
+
+    Returns:
+
+    """
+
+    info = nib.load(nii_path)
+    image = np.asanyarray(info.dataobj)
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    json_path = nii_path.split('.nii')[0] + '.json'
+
+    if os.path.isfile(json_path):
+        json_data = json.load(open(json_path))
+    else:
+        raise ValueError('Missing json file')
+
+    # NOTE: nib.load automatically scales the nifti and replaces scl_inter and scl_slope with 'nan'
+    # More info in the "Data scaling" section in https://nipy.org/nibabel/nifti_images.html
+    if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens') \
+            and (image_type(json_data) == 'phase'):
+
+        image, info = convert_siemens_phase(image, info)
+
+    return image, info, json_data
+
+
+def image_type(json_data):
```

Comment:
json_data is misleading in Python since we can use a fairly complete JSON package. You'll surely have people trying to use this function with json type variables (string). You should check if it is a string from the JSON package and load it:
```suggestion
def image_type(json_data):
    if isinstance(json_data, string):
        data = json.loads(json_data)
```
Something like this.

Reply:
I just changed the variable description because I think it's better to only have on type of variable for this and also because json_data is already returned by` json_data = json.load(open(json_path))` in `read_nii`